### PR TITLE
Support trailing commas in sbt files

### LINKED ISSUE #1098: Support trailing commas in sbt files
Currently, the sbt dialect is based of 211, which does not permit trailing commas. We should base it off the default lenient dialect.

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #1098

Please review mercilessly, I'm happy to iterate.

### CODE REVIEW

File: scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala [hunk 1]
```diff
@@ -89,6 +89,6 @@ object ScalafmtRunner {
     */
   val statement = default.copy(parser = scala.meta.parsers.Parse.parseStat)
 
-  val sbt = default.copy(dialect = scala.meta.dialects.Sbt0137)
+  val sbt = default
```
Comment (olafurpg) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/config/ScalafmtRunner.scala]: Can we do `forSbt` here?

File: scalafmt-core/shared/src/test/scala/org/scalafmt/ScalafmtRunnerTest.scala [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package org.scalafmt
+
+import scala.meta._
+import org.scalafmt.config._
+import org.scalatest.FunSuite
+
+class ScalafmtRunnerTest extends FunSuite {
+  test("sbt dialect supports trailing commas") {
+    val code = ScalafmtRunner.sbt.forSbt
```
Comment (olafurpg) [inline scalafmt-core/shared/src/test/scala/org/scalafmt/ScalafmtRunnerTest.scala]: `.sbt.forSbt` looks weird

File: scalafmt-core/shared/src/test/scala/org/scalafmt/ScalafmtRunnerTest.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          |  a,
+          |
+          |
+          |  b,
+          |)
+        """.stripMargin
+      )
+      .parse[Source]
+      .get
+    assert(code != null)
```
Comment (olafurpg) [inline scalafmt-core/shared/src/test/scala/org/scalafmt/ScalafmtRunnerTest.scala]: Remove, parse will never return null

Comment (dwijnand) [reply scalafmt-core/shared/src/test/scala/org/scalafmt/ScalafmtRunnerTest.scala]: I thought I needed this to make the types align, but guess with scalatest.FunSuite you don't. Done.

**Pull request merged**